sui_core/src/authority/authority_store.rs

```diff
@@ -142,9 +149,12 @@ pub fn open_cf_opts<P: AsRef<Path>>(
     Ok(rocksdb)
 }
 
-impl<const ALL_OBJ_VER: bool> SuiDataStore<ALL_OBJ_VER> {
+impl<const ALL_OBJ_VER: bool, S: AuthoritySignInfoTrait> SuiDataStore<ALL_OBJ_VER, S>
```

Comment:
How about replacing `AuthoritySignInfoTrait` with its constituent traits (which I suspect may be a bit too strong: my hunch is `Eq + Serialize + for<'de> Deserialize<'de>` would work)?